src/plugins/intel_gpu/include/intel_gpu/primitives/primitive.hpp

```diff
@@ -213,6 +213,30 @@ struct primitive_info {
     CLDNN_DEFINE_TYPE_ID(PType)              \
     CLDNN_DEFINE_TYPE_STRING(PType)
 
+#define CLDNN_DEFINE_PRIMITIVE_TYPE_ID(PType)           \
+    primitive_type_id PType::type_id() {                \
+        static primitive_type_base<PType> instance;     \
+        return &instance;                               \
+    }                                                   \
+    bool _##PType##_added_ = prim_map_storage::instance().set_type_id(#PType, PType::type_id());
+
+struct prim_map_storage {
+    static prim_map_storage& instance() {
+        static prim_map_storage instance;
+        return instance;
+    }
+
+    const cldnn::primitive_type_id get_type_id(const std::string& type_string) const {
+        return map.at(type_string);
+    }
+
+    bool set_type_id(const std::string& type_string, const cldnn::primitive_type_id type_id) {
+        return map.insert({type_string, type_id}).second;
+    }
+
+private:
+    std::unordered_map<std::string, cldnn::primitive_type_id> map;
```

Comment:
Maybe we can make this map a static field of primitive_type and insert type_id in primitive_type_base<PType> c-tor?

Reply:
If the point to insert type_id is moved to c-tor, primitives that have not yet been created will not exist in the map. Then, when deserializing, the type_id cannot be obtained by type name, which causes a problem.

Reply:
primitive_type objects are static, so all primitives are supposed to be initialized on app startup, aren't they?

Reply:
After adding a ctor in `primitive_type_base`, I checked the execution step by debugger.
In my test, the first calling point was not on app startup as below.
![image](https://user-images.githubusercontent.com/89953545/202465443-845989bc-5962-4180-9e30-4ae37949b48d.png)